torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2827,11 +2876,30 @@ def get_symbol(tensor_dim_src):
                 concrete_val = self.evaluate_expr(sympy.Eq(s1, s2))
                 if not concrete_val:
                     raise ConstraintViolationError(
-                        f"{src1.name()} = {self.var_to_val[s1]}"
+                        f"{src1.name()} = {s1.subs(self.var_to_val)}"
                         " is not equal to "
-                        f"{src2.name()} = {self.var_to_val[s2]}"
+                        f"{src2.name()} = {s2.subs(self.var_to_val)}"
                     )
 
+            for src, root, fn in equalities_inputs.derived_equalities:
+                s1 = get_symbol(src)
+                s2 = root if isinstance(root, sympy.Symbol) else get_symbol(root)
+                s2_ = fn(s2)
+                concrete_val = self.evaluate_expr(sympy.Eq(s1, s2_))
+                if not concrete_val:
+                    # we were unable to prove that src = fn(root) holds
```

Comment:
So, supposing you DO use evaluate_expr, here, not only were you unable to prove that src = fn(root) didn't hold, but in fact, if concrete_val is False, the user has given us inputs for which the constraint was expressly violated!

Reply:
Yup! So we will emit a `ConstraintViolationError` here.

Reply:
See above.